packages/turf-boolean-contains/index.js

```diff
@@ -0,0 +1,316 @@
+        }
+    }
+    return output;
+}
+
+function isMultiPointInMultiPoint(multiPoint1, multiPoint2) {
+    var foundAMatch = 0;
+    for (var i = 0; i < multiPoint2.coordinates.length; i++) {
+        var anyMatch = false;
+        for (var i2 = 0; i2 < multiPoint1.coordinates.length; i2++) {
+            if (compareCoords(multiPoint2.coordinates[i], multiPoint1.coordinates[i2])) {
+                foundAMatch++;
+                anyMatch = true;
+                break;
+            }
+        }
+        if (!anyMatch) {
+            return false;
+        }
+    }
+    return foundAMatch > 0;
+}
+
+// http://stackoverflow.com/a/11908158/1979085
+function isPointOnLine(lineString, point) {
+    var output = false;
+    for (var i = 0; i < lineString.coordinates.length - 1; i++) {
+        var incEndVertices = true;
+        if (i === 0 || i === lineString.coordinates.length - 2) {
+            incEndVertices = false;
+        }
+        if (isPointOnLineSegment(lineString.coordinates[i], lineString.coordinates[i + 1], point.coordinates, incEndVertices)) {
+            output = true;
+            break;
+        }
+    }
+    return output;
+}
+
+function isMultiPointOnLine(lineString, multiPoint) {
+    var output = true;
+    for (var i = 0; i < multiPoint.coordinates.length; i++) {
+        var pointIsOnLine = false;
+        for (var i2 = 0; i2 < lineString.coordinates.length - 1; i2++) {
+            var incEndVertices = true;
+            if (i2 === 0 || i2 === lineString.coordinates.length - 2) {
+                incEndVertices = false;
+            }
+            if (isPointOnLineSegment(lineString.coordinates[i2], lineString.coordinates[i2 + 1], multiPoint.coordinates[i], incEndVertices)) {
+                pointIsOnLine = true;
+                break;
+            }
+        }
+        if (!pointIsOnLine) {
+            output = false;
+            break;
+        }
+    }
+    return output;
+}
+
+function isMultiPointInPoly(polygon, multiPoint) {
+    var output = true;
+    for (var i = 0; i < multiPoint.coordinates.length; i++) {
+        var isInside = inside(multiPoint.coordinates[1], polygon);
+        if (!isInside) {
+            output = false;
+            break;
+        }
+    }
+    return output;
+}
+
+function isLineOnLine(lineString1, lineString2) {
+    var output = true;
+    for (var i = 0; i < lineString2.coordinates.length; i++) {
+        var checkLineCoords = isPointOnLine(lineString1, {type: 'Point', coordinates: lineString2.coordinates[i]});
+        if (!checkLineCoords) {
+            output = false;
+            break;
+        }
+    }
+    if (output) {
+        if (compareCoords(lineString1.coordinates[0], lineString2.coordinates[0]) ||
+            compareCoords(lineString1.coordinates[lineString1.coordinates.length - 1], lineString2.coordinates[lineString2.coordinates.length - 1])) {
+            output = false;
+        }
+    }
+    return output;
+}
+
+function isLineInPoly(polygon, linestring) {
```

Comment:
@rowanwins Same issue as above. `contains` only requires that some line point is in the polygon interior, not necessarily a vertex. You can have a line in which vertices are entirely on the boundary but the line is still contained.